plugin/src/main/groovy/com/novoda/gradle/command/DemoModeTask.groovy

```diff
@@ -0,0 +1,64 @@
+package com.novoda.gradle.command
+
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.tasks.TaskAction
+
+class DemoModeTask extends AdbTask {
+
+    boolean enable
+    NamedDomainObjectContainer<DemoModeExtension> commands
+
+    DemoModeTask() {
+        this.group = AndroidCommandPlugin.TASK_GROUP
+        this.description = "${enable ? 'Enables' : 'Disables'} demo mode on the connected device"
+    }
+
+    @TaskAction
+    void exec() {
+        assertDeviceConnected()
+        assertDeviceAndroidM()
+        if (enable) {
+            allowDemoMode()
+            executeDefaults()
+            commands.each {
+                broadcast it.name, it.extras
+            }
+        } else {
+            broadcast 'exit'
+        }
+    }
+
+    private void executeDefaults() {
+        broadcastDefault 'network', [mobile: 'show', level: '4']
+        broadcastDefault 'battery', [level: '100', plugged: 'false']
+        broadcastDefault 'network', [wifi: 'show', level: '4']
+        broadcastDefault 'clock', [hhmm: '0810']
+        broadcastDefault 'notifications', [visible: 'false']
+    }
+
+    private void allowDemoMode() {
+        runCommand(['shell', 'settings', 'put', 'global', 'sysui_demo_allowed', '1'])
+    }
+
+    def broadcastDefault(String name, intentExtras) {
+        if (!commands.findByName(name)) {
+            broadcast(name, intentExtras)
+        }
+    }
+
+    def broadcast(String name, Map<String, String> intentExtras = [:]) {
+        def args = intentExtras.collectMany { key, value -> ['-e', "$key", "$value"] }
+        runCommand([
+                'shell', 'am', 'broadcast',
+                '-a', 'com.android.systemui.demo',
+                '-e', 'command', name,
+                *args
+        ])
+    }
+
+    void assertDeviceAndroidM() {
+        if (device().sdkVersion() < 23) {
+            logger.warn "Connected device must have at least Android Marshmallow (API level 23) to enable Demo Mode"
```

Comment:
Here I decided to just warn the user that the demo mode may not work as expected. I first made the Gradle execution fail but that is too much. Its because this task may run before other tasks and we don't want to fail the whole execution.

![screen shot 2018-01-26 at 10 22 20 am](https://user-images.githubusercontent.com/763339/35433211-df42d47c-0282-11e8-8314-65d54310b49b.png)

Also in some Lollipop devices (mostly Nexuses) these `broadcast` commands actually enable demo mode even though Google says it is introduced in Marshmallow.